src/Adapter/ReactPhp/EventLoop.php

```diff
@@ -22,7 +22,7 @@ public function wait(Promise $promise)
         $value = null;
         $isRejected = false;
         $promiseSettled = false;
-        Internal\PromiseWrapper::downcast($promise)->getReactPromise()->then(
+        Internal\PromiseWrapper::downcast($promise)->markYielded()->getReactPromise()->then(
```

Comment:
Same here, we could use something like a `Internal\PromiseWrapper::wait` function, to ensure that `markYielded` is not used wrongly elsewhere.

(and same in other adapter)